metrics/rules/aip140_test.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2020 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rules
+
+import (
+	"testing"
+)
+
+func TestSnakeCase(t *testing.T) {
+	if pass, suggestion := snakeCase("helloWorld"); pass {
+		t.Error("Given \"helloWorld\", snakeCase() returned true, expected false")
```

Comment:
These tests seem to replicate a lot of code. I think it would be helpful to restructure them to start by defining an array of test strings and then looping over those strings calling the code in lines 22-26 (for this function, similarly for the others). That would make it easier to add new patterns to each test or to change common code used by all the tests.